pytorch_lightning/callbacks/lambda_function.py

```diff
@@ -78,6 +78,9 @@ def __init__(
         on_exception: Optional[Callable] = None,
         on_save_checkpoint: Optional[Callable] = None,
         on_load_checkpoint: Optional[Callable] = None,
+        on_load_checkpoint_new: Optional[Callable] = None,
+        state_dict: Optional[Callable] = None,
+        load_state_dict: Optional[Callable] = None,
```

Comment:
I don't think this should be added to the lambda function callback. Almost by definition, the lambda callback is for purely functional use cases. There shouldn't be any state to persist